src/lib.rs

```diff
@@ -687,26 +699,69 @@ impl EGraph {
 
         let mut buf = String::new();
         let s = &mut buf;
-        s.push_str("(\n");
-        if terms.is_empty() {
-            log::info!("   (none)");
+        if !is_csv {
+            s.push_str("(\n");
+            if terms.is_empty() {
+                log::info!("   (none)");
+            }
         }
+
         for (term, output) in terms.iter().zip(&outputs.unwrap()) {
-            let tuple_str = format!(
-                "   {}{}",
-                termdag.to_string(term),
-                if !out_is_unit {
-                    format!(" -> {}", termdag.to_string(output))
-                } else {
-                    "".into()
-                },
-            );
-            log::info!("{}", tuple_str);
-            s.push_str(&tuple_str);
+            if is_csv {
+                match term {
+                    Term::App(name, children) => {
+                        debug_assert!(name == sym);
+                        s.push_str(name);
+                        for child in children
+                            .iter()
+                            .map(|child| termdag.to_string(termdag.get(*child)))
+                        {
+                            s.push(',');
```

Comment:
Might be worth using a dedicated CSV exporter to handle escaping more easily in the future. https://github.com/BurntSushi/rust-csv